dbt/include/spark/macros/adapters.sql

```diff
@@ -16,9 +16,10 @@
 
 {% macro spark__location_clause() %}
   {%- set location_root = config.get('location_root', validator=validation.any[basestring]) -%}
+  {%- set schema = model['schema'] -%}
   {%- set identifier = model['alias'] -%}
   {%- if location_root is not none %}
-    location '{{ location_root }}/{{ identifier }}'
+  location '{{ location_root }}/{{ schema }}/{{ identifier }}'
```

Comment:
An interesting way to make this non-breaking would be to make this configurable:
```suggestion
 {%- set use_schema_in_location = config.get('use_schema_in_location_root') -%}
 {%- if location_root is not none %}
   {%- if use_schema_in_location is not none %}
     location '{{ location_root }}/{{ schema }}/{{ identifier }}'
    {% else %}
     location '{{ location_root }}/{{ identifier }}'
```

Reply:
Couldn't we also update `config.get()` to stuff the schema into the location root if it's present?

Reply:
@JCZuurmond do you have an opinion on that?

Reply:
Yes, I would like to have the location with the schema enabled by default. 

Without the schema, users may overwrite each other's tables even when they have them in separate schemas because they are overwriting the underlying data via the location. This happened to colleagues of mine, which is a hard bug to track down.

I discussed the non-breaking/breaking change with @jtcohen6. I recall we were ok with it, but I do not see this conversation back in the [issue](#239) maybe we had the conversation over Slack. 

In any case, the change is _kinda_ of a breaking change. The location is added when a user creates a new table; either because the data model is new (in the target schema), or when the materialization is `table`, or when using `--full-refresh`. Existing incremental tables are not affected and for the other options the tables are recreated anyway.

My preference would be to enable the location with schema by default AND communicate clearly what is changed and why. It will be confusing if users do not know/understand this change and they see that some of the (older) tables have a location without schema and others have a location with schema. However, it will be even more confusing when multiple developers work together within the same warehouse and they are interfering with each other's tables unknowingly.